imports/timer/shared.lua

```diff
@@ -0,0 +1,118 @@
+---@class TimerPrivateProps
+---@field onEnd? fun() cb function triggered when the timer finishes
+---@field async? boolean wether the timer should run asynchronously or not
+---@field startTime number the gametimer stamp of when the timer starts. changes when paused and played
+---@field triggerOnEnd boolean set in the forceEnd method using the optional param. wether or not the onEnd function is triggered when force ending the timer early
+---@field currentTime number current timer length
+---@field tickActive boolean or not the tick is running
+
+---@class CTimer : OxClass
```

Comment:
Since LLS classes are applied globally I try to avoid potential conflicts by using Ox for classes (i.e. OxTimer).